packages/controls/src/widget_dragdrop.ts

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import {
+    CoreDOMWidgetModel
+} from './widget_core';
+
+import {
+    DOMWidgetView, unpack_models, WidgetModel, WidgetView, JupyterLuminoPanelWidget, reject
+} from '@jupyter-widgets/base';
+
+import $ from 'jquery';
+
+export
+class DraggableBoxModel extends CoreDOMWidgetModel {
+    defaults(): Backbone.ObjectHash {
+        return {
+            ...super.defaults(),
+            _view_name: 'DraggableBoxView',
+            _model_name: 'DraggableBoxModel',
+            child: null,
+            draggable: true,
+            drag_data: {}
+        };
+    }
+
+    static serializers = {
+        ...CoreDOMWidgetModel.serializers,
+        child: {deserialize: unpack_models}
+    };
+}
+
+export
+class DropBoxModel extends CoreDOMWidgetModel {
+    defaults(): Backbone.ObjectHash {
+        return {
+            ...super.defaults(),
+            _view_name: 'DropBoxView',
+            _model_name: 'DropBoxModel',
+            child: null
+        };
+    }
+
+    static serializers = {
+        ...CoreDOMWidgetModel.serializers,
+        child: {deserialize: unpack_models}
+    };
+}
+
+class DragDropBoxViewBase extends DOMWidgetView {
+    child_view: DOMWidgetView | null;
+    pWidget: JupyterLuminoPanelWidget;
+
+    _createElement(tagName: string): HTMLElement {
+        this.pWidget = new JupyterLuminoPanelWidget({ view: this });
+        return this.pWidget.node;
+    }
+
+    _setElement(el: HTMLElement): void {
+        if (this.el || el !== this.pWidget.node) {
+            // Boxes don't allow setting the element beyond the initial creation.
+            throw new Error('Cannot reset the DOM element.');
+        }
+        this.el = this.pWidget.node;
+        this.$el = $(this.pWidget.node);
+    }
+
+    initialize(parameters: WidgetView.IInitializeParameters): void {
+        super.initialize(parameters);
+        this.add_child_model(this.model.get('child'));
+        this.listenTo(this.model, 'change:child', this.update_child);
+
+        this.pWidget.addClass('jupyter-widgets');
+        this.pWidget.addClass('widget-container');
+        this.pWidget.addClass('widget-draggable-box');
```

Comment:
Is this class needed? It doesn't seem to be used.

Reply:
I think I just tried to follow conventions: https://github.com/jupyter-widgets/ipywidgets/blob/7cb20cdf2df8ad54fbc0897934b3075998a9a9b5/packages/controls/src/widget_box.ts#L83-L85 

someone who wants to style the drag drop widgets might want to add CSS to this?
On the other hand, the two leaf classes should probably add another class as well.

Reply:
> someone who wants to style the drag drop widgets might want to add CSS to this?

That could indeed be useful.

For the case of the widget_box, there is actually some default css:

https://github.com/jupyter-widgets/ipywidgets/blob/7cb20cdf2df8ad54fbc0897934b3075998a9a9b5/packages/controls/css/widgets-base.css#L98-L103